Initial checks for onboarding page

Vivek205 opened the pull request.

tiero commented:
please review @raamb

ferrouswheel commented:
Not that the parseServiceSpec is in every UI component, so changes will have to propagate.

There is also a a bug in it, that I had to fix for #156

This is a good highlight for why we putting off refactoring the custom UI components #79 is a bad idea. It's going to be impossible to reconcile changes between all the individual components.

ferrouswheel commented on src/components/service/DefaultService.js:
```diff
@@ -25,14 +30,24 @@ export default class DefaultService extends React.Component {
+                    return
                 }
+
+                this.state.response = JSON.stringify(nextProps.response || nextProps.responseObject)
```
Please don't stringify the response by default. UI components need access to the underlying structure of the messages and extra serialisation steps makes things frustrating.

tiero replied:
you are right. This was my attempt to fix a crash when there is a malformed response. I will revert the code.

ferrouswheel commented:
@raamb ... uh, can we wait until the PR is fixed before merging these things?

tiero commented:
> Not that the parseServiceSpec is in every UI component, so changes will have to propagate.

Do you mean to replace parseServiceSpec with the `protobuf.js` utility? I left to make it backward compatible and not start a refactor and keep this change very isolated.  

> There is also a a bug in it, that I had to fix for #156
I don't understand what I should do 🤔 

> This is a good highlight for why putting off refactoring the custom UI components #79 is a bad idea. It's going to be impossible to reconcile changes between all the individual components.
Agreed

ferrouswheel commented:
The bug is that we are assigning directly to the component state instead of using `setState`.

But yeah, the whole thing needs to be majorly changed (`componentWillRecieveProps` is deprecated in recent version of react, because people like us misuse it).

tiero commented:
> The bug is that we are assigning directly to the component state instead of using `setState`.
> 
> But yeah, the whole thing needs to be majorly changed (`componentWillRecieveProps` is deprecated in recent version of react, because people like us misuse it).

Yes at first glance even to me that was very strange, but if it was in master I assumed was ok or for some reason needed. (Isn't very common? 😂) 

Will fix in #164

raamb commented:
Guys. I am working with a react expert to come up with a workable
solution.  Should have it in a day or two. We are trying  to resolve this
via composition rather than inheritance.

Thanks,
Raam

On Thu, Feb 14, 2019, 03:45 Marco Argentieri <notifications@github.com
wrote:

> The bug is that we are assigning directly to the component state instead
> of using setState.
>
> But yeah, the whole thing needs to be majorly changed (
> componentWillRecieveProps is deprecated in recent version of react,
> because people like us misuse it).
>
> Yes at first glance even to me that was very strange, but if it was in
> master I assumed was ok or for some reason needed. (Isn't very common? 😂)
>
> Will fix in #164 <https://github.com/singnet/snet-dapp/pull/164>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/singnet/snet-dapp/pull/152#issuecomment-463397651>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApymOVSVFX2DrS3cKF7rbci418aXKgCvks5vNI6TgaJpZM4a2Pzt>
> .
>

Vivek205 committed:
Merging latest change from main repo (#96)

* typo: legnth -> length

* removed username from updateProfile

* deleted commented code

* syncing the props with local state

* Arranged imports

* renamed: generateAPIInit -> initializeAPIOptions

* Move example service to new directory structure

* renamed: myInit -> apiOptions

* removed the componentDidUpdate

* Fix lint issues

* handled empty feedback

* cntk img recognition using sdk

* getValue()

* serviceRequestComplete handler should be at the last

* moved cntkImgRecon to appropriate dir heirarchy

* stubs with namespace

* using snakeCase for directory

* updating the import 

Updating the import of `const CNTKImageRecognition = lazy(() => import("./snet/cntk-image-recon"));` -> `const CNTKImageRecognition = lazy(() => import("./snet/cntk_image_recon"));` to fix the code_build.

* sdk integration CNTK-Language-understanding

* conflict solved for index,js

* Error Boundary for ThirdPartyAIServices

* moved org_id with snet to snet folder

* error boundary for AI services

* stubs for face-align

* setup request for face-align

* loader and reset-run

* isComplete and reset-run lang_understanding

* parsing response face-align

* isComplete and reset-run face_align

* removed unused <p>

* zeta36 chess and alpha is moved to new folder inside snet

* Metering service endpoint setup

* fetching metering data

* added eslint disable on chess-service stubs

* typo in APIEndpoints

* parsing the StarRatingComponent's props

* created purchase tab and its content for demo example.added styling for info box

* deleted unused props

* Purchase Toggler

* computeActiveSection

* freeCallsRemaining

* added two components active session and expired session for purcahse

* UserProfile Account

* Grid for Account

* props to StyledDropDown

* Free/Regular calls

* error boundary

* metadata for free call

* alertbox: lint

* getMethodNAmes from prop

* ActiveSession - Expired Session

* created component for accounts tab for user and styled elements inside it

* Metering API

* error handling on End

* free-trial-testing changes

* Simplify metadata generation

* Make getMethods private

* fetchAuthToke -> fetchAuthToken

* Remove duplicate code

* uncomment fetFreeCall and reset purchaseCompleted

* returning base64 string - signature

* lint:fix

* fetching user Details - email & token

* passing emailId to metering service

* reset-run example-service

* /:serviceRowId -> org/:a/service/:/b

* extracted fetchAuthUser

* metering: user_id -> username

* commenting pricingStrategy

* serviceMetaData

* added avatar in the cardheader of service list view

* added avatar in the cardheader of service card view

* fetchAuthUser

* added avatar in the service details page

* Subheader

* classes

* added rating section inside the cardheader of service list view

* facilitate serviceList -> details flow

* removed redundant thro

* MeteringDataSuccess

* moved: freeCallsRemaining to redux

* freeCallsAllowed

* lint:fix

* using dappUser gateway for metering service

* alert Link

* renamed fetchAuthUser -> fetchAuthenticatedUser

* removed the {"  "}

* created channelSelectionNox component

* moved serviceMetadata  -> serviceDetailsReducer

* added a reactComponent for channelSelectionBox

* moved freeCalls -> serviceDetailsReducer

* waiting for fetchMeteringData in Purchase </>

* extracted demoProgressStatus -> enum

* included isTermsAccepted in userReducer.

* updating the isTermsAccepted flag

* userProfileAPI call -> updating isTermsAccepted

* update isWalletAssigned and isTermsAccepted

* eslint objectShorthand

* created and styled paymentinfocard

* added buttons to purchase flow

* handleSubmit -> TermsofUse

* disabled styling to the channelSelecionBox

* added multiple and single call box

* disabled input in single call

* New login - registration flow

* isWalletAssigned -> walletAddress

* walletAddress -> wallet

* workAround to accomodate /profile API

* created and styled the dialog box

* eslint: fixes

* removed explicit true on props

* use es6 for methods

* reverting ActiveSession component

* removed serviceMetadata from serviceReducer

* added avatar to list view

* checking for metamask availability in expiredSession

* Add defaults for sandbox mode

* added styling for selectedChannelBox

* cursor: pointer for channelSelectionBox

* reverse the alternate categoried

* MetmaskFlow

* DialogTitle

* No of Calls

* inputProps

* CallPriceUnit

* exponential -> string

* PaymentInfoCardData

* Deposit

* Move third party service custom ui to new structure

* Async Calls for Purchase

* fetch escrow/channel balance

* unused async in funcs

* using ===

* unused async

* fix:bug

* eslint: fixes

* forgotPassword fix

* corrected the img path

* Add lodash

* Parse group info

* added a dummy image for fetaure Media of get started

* removed duplicate fetchAuthenticatedUser

fetchAuthenticatedUser was already declared, duplicate was added while resolving conflict. removed it.

* Fix pricing strategy

* eslint: fixes

* added dummy video and gif file to getStarted

* connect MetaMask

* connect Metamask

* removed unwanted console

* added service creator avatar in service details and service card view

* adjusted the placemnet of service creator avatar inservice card view

* updated eslint

* Fixed the links in the footer

* removed unwanted import

* removed pre-tag in alertBox

* private route for userProfile

* added video tag for mp4 and img tag for gif in get started

* walletType in userProfile account

* wallet should be empty in initialReducer

* eslint:fixees

* content || placeholder

* added new logo

* accounts page

* bug fix

* Fetch escrow balance

* Copy edit changes as suggested by Marketing team

* Fixed copy based

* removed all the font family and added Muli as default

* deposit and withdraw

* Loader and Error handlers

* Fetch payment channel details

* Show escrow and channel balance

* Add fund channel to channel management

* Update snet-sdk-web version

* Add missing PaymentManagementStrategy

* added font weight accordingly

* hardcoded Payment Endpoints

* added font weight accordingly

* changed the position of the rating div in list view

* footer logo size given, hover effect for links

* pagination styling corrected

* changed the default service image

* signup page white space issue resolved

* hover state for forgot password link is corrected

* hover state for forgot password link is corrected

* adjusted progress bar on onboarding screen

* removed the white space in forgot password screen

* get started will open in same tab, added header and footer to get stared

* added new font to styledDropdown

* removed unwanted console and imports

Vivek205 committed:
Extend and Add Funds  (#97)

* renamed: generateAPIInit -> initializeAPIOptions

* Move example service to new directory structure

* renamed: myInit -> apiOptions

* removed the componentDidUpdate

* Fix lint issues

* handled empty feedback

* cntk img recognition using sdk

* getValue()

* serviceRequestComplete handler should be at the last

* moved cntkImgRecon to appropriate dir heirarchy

* stubs with namespace

* using snakeCase for directory

* updating the import 

Updating the import of `const CNTKImageRecognition = lazy(() => import("./snet/cntk-image-recon"));` -> `const CNTKImageRecognition = lazy(() => import("./snet/cntk_image_recon"));` to fix the code_build.

* sdk integration CNTK-Language-understanding

* conflict solved for index,js

* Error Boundary for ThirdPartyAIServices

* moved org_id with snet to snet folder

* error boundary for AI services

* stubs for face-align

* setup request for face-align

* loader and reset-run

* isComplete and reset-run lang_understanding

* parsing response face-align

* isComplete and reset-run face_align

* removed unused <p>

* zeta36 chess and alpha is moved to new folder inside snet

* Metering service endpoint setup

* fetching metering data

* added eslint disable on chess-service stubs

* typo in APIEndpoints

* parsing the StarRatingComponent's props

* created purchase tab and its content for demo example.added styling for info box

* deleted unused props

* Purchase Toggler

* computeActiveSection

* freeCallsRemaining

* added two components active session and expired session for purcahse

* UserProfile Account

* Grid for Account

* props to StyledDropDown

* Free/Regular calls

* error boundary

* metadata for free call

* alertbox: lint

* getMethodNAmes from prop

* ActiveSession - Expired Session

* created component for accounts tab for user and styled elements inside it

* Metering API

* error handling on End

* free-trial-testing changes

* Simplify metadata generation

* Make getMethods private

* fetchAuthToke -> fetchAuthToken

* Remove duplicate code

* uncomment fetFreeCall and reset purchaseCompleted

* returning base64 string - signature

* lint:fix

* fetching user Details - email & token

* passing emailId to metering service

* reset-run example-service

* /:serviceRowId -> org/:a/service/:/b

* extracted fetchAuthUser

* metering: user_id -> username

* commenting pricingStrategy

* serviceMetaData

* added avatar in the cardheader of service list view

* added avatar in the cardheader of service card view

* fetchAuthUser

* added avatar in the service details page

* Subheader

* classes

* added rating section inside the cardheader of service list view

* facilitate serviceList -> details flow

* removed redundant thro

* MeteringDataSuccess

* moved: freeCallsRemaining to redux

* freeCallsAllowed

* lint:fix

* using dappUser gateway for metering service

* alert Link

* renamed fetchAuthUser -> fetchAuthenticatedUser

* removed the {"  "}

* created channelSelectionNox component

* moved serviceMetadata  -> serviceDetailsReducer

* added a reactComponent for channelSelectionBox

* moved freeCalls -> serviceDetailsReducer

* waiting for fetchMeteringData in Purchase </>

* extracted demoProgressStatus -> enum

* included isTermsAccepted in userReducer.

* updating the isTermsAccepted flag

* userProfileAPI call -> updating isTermsAccepted

* update isWalletAssigned and isTermsAccepted

* eslint objectShorthand

* created and styled paymentinfocard

* added buttons to purchase flow

* handleSubmit -> TermsofUse

* disabled styling to the channelSelecionBox

* added multiple and single call box

* disabled input in single call

* New login - registration flow

* isWalletAssigned -> walletAddress

* walletAddress -> wallet

* workAround to accomodate /profile API

* created and styled the dialog box

* eslint: fixes

* removed explicit true on props

* use es6 for methods

* reverting ActiveSession component

* removed serviceMetadata from serviceReducer

* added avatar to list view

* checking for metamask availability in expiredSession

* Add defaults for sandbox mode

* added styling for selectedChannelBox

* cursor: pointer for channelSelectionBox

* reverse the alternate categoried

* MetmaskFlow

* DialogTitle

* No of Calls

* inputProps

* CallPriceUnit

* exponential -> string

* PaymentInfoCardData

* Deposit

* Move third party service custom ui to new structure

* Async Calls for Purchase

* fetch escrow/channel balance

* unused async in funcs

* using ===

* unused async

* fix:bug

* eslint: fixes

* forgotPassword fix

* corrected the img path

* Add lodash

* Parse group info

* added a dummy image for fetaure Media of get started

* removed duplicate fetchAuthenticatedUser

fetchAuthenticatedUser was already declared, duplicate was added while resolving conflict. removed it.

* Fix pricing strategy

* eslint: fixes

* added dummy video and gif file to getStarted

* connect MetaMask

* connect Metamask

* removed unwanted console

* added service creator avatar in service details and service card view

* adjusted the placemnet of service creator avatar inservice card view

* updated eslint

* Fixed the links in the footer

* removed unwanted import

* removed pre-tag in alertBox

* private route for userProfile

* added video tag for mp4 and img tag for gif in get started

* walletType in userProfile account

* wallet should be empty in initialReducer

* eslint:fixees

* content || placeholder

* added new logo

* accounts page

* bug fix

* Fetch escrow balance

* Copy edit changes as suggested by Marketing team

* Fixed copy based

* removed all the font family and added Muli as default

* deposit and withdraw

* Loader and Error handlers

* Fetch payment channel details

* Show escrow and channel balance

* Add fund channel to channel management

* Update snet-sdk-web version

* Add missing PaymentManagementStrategy

* added font weight accordingly

* hardcoded Payment Endpoints

* added font weight accordingly

* changed the position of the rating div in list view

* footer logo size given, hover effect for links

* pagination styling corrected

* changed the default service image

* signup page white space issue resolved

* hover state for forgot password link is corrected

* hover state for forgot password link is corrected

* adjusted progress bar on onboarding screen

* removed the white space in forgot password screen

* get started will open in same tab, added header and footer to get stared

* added new font to styledDropdown

* extendAndAddFunds

* removed unwanted console and imports

* handle channelBalance == 0

* returnToPurchase on ResetAndRun

* initializePaymentChannel

Vivek205 committed:
fetching paymnet details from serviceMetadata

Vivek205 committed:
Merge branch 'master' into feature-service-details-purchase-flow

Vivek205 committed:
initial checks for onboarding

Vivek205 committed:
Merge branch 'master' into feature-service-details-purchase-flow
